sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -267,23 +230,8 @@ public virtual async Task<Response<UploadManifestResult>> UploadManifestAsync(St
             scope.Start();
             try
             {
-                using MemoryStream stream = await BufferStreamAsync(manifestStream, true).ConfigureAwait(false);
-                string digest = null;
-
-                string tagOrDigest = options.Tag ?? (digest = OciBlobDescriptor.ComputeDigest(stream));
-                ResponseWithHeaders<ContainerRegistryCreateManifestHeaders> response = await _restClient.CreateManifestAsync(_repositoryName, tagOrDigest, manifestStream, ManifestMediaType.OciManifest.ToString(), cancellationToken).ConfigureAwait(false);
-
-                bool valid = digest != null ?
-                    ValidateDigest(digest, response.Headers.DockerContentDigest) :
-                    ValidateDigest(stream, response.Headers.DockerContentDigest);
-
-                if (!valid)
-                {
-                    throw _clientDiagnostics.CreateRequestFailedException(response,
-                        new ResponseError(null, "The digest in the response does not match the digest of the uploaded manifest."));
-                }
-
-                return Response.FromValue(new UploadManifestResult(response.Headers.DockerContentDigest), response.GetRawResponse());
+                using Stream stream = await CopyStreamAsync(manifestStream, true).ConfigureAwait(false);
```

Comment:
So we do buffer the stream into memory for manifests, but are these usually smaller than blobs?

Reply:
Yes, I'm doing that so we don't dispose the caller's passed-in stream.